core/arch/arm/plat-imx/drivers/tzc380.c

```diff
@@ -2,36 +2,106 @@
 /*
  * Copyright 2019 Pengutronix
  * All rights reserved.
+ * Copyright 2019 NXP
  *
  * Rouven Czerwinski <entwicklung@pengutronix.de>
  */
 
 #include <drivers/tzc380.h>
 #include <imx-regs.h>
 #include <imx.h>
+#include <kernel/panic.h>
 #include <mm/core_memprot.h>
 #include <mm/generic_ram_layout.h>
 
+#if (defined(PLATFORM_FLAVOR_mx6solosabresd) \
+	|| defined(PLATFORM_FLAVOR_mx6solosabreauto) \
+	|| defined(PLATFORM_FLAVOR_mx6sxsabresd) \
+	|| defined(PLATFORM_FLAVOR_mx6sxsabreauto) \
+	|| defined(PLATFORM_FLAVOR_mx6ulevk) \
+	|| defined(PLATFORM_FLAVOR_mx6ul9x9evk) \
+	|| defined(PLATFORM_FLAVOR_mx6ullevk) \
+	|| defined(PLATFORM_FLAVOR_mx7dsabresd))
```

Comment:
I don't think its a good idea to couple the TZASC configuration to board flavours instead of the CPUs. Several boards remain unsupported with this list. Not a fan of the `#ifdef` in `.c` files and code duplication that this code introduces either.

Reply:
I think It's one way of doing it.
I thought about sorting TZASC configurations by CPUs but since there are disparities within IMX6 (TZASC and TZASC2) and between IMX6/7 and IMX8 I went for this implementation.
Sorting it by platform flavors might duplicate code, I admit, but I think it's clearer.
I only added boards I could test myself.
Since there are two TZASC controllers in imx6q series, you are going to end up anyway with #ifdef in .c files I believe.

Reply:
No, please take a look at my recently updated PR #3017 which uses an offset for the second TZASC. This can also easily cover the i.MX8 platform by introducing an offset which is only set for the I.MX8 platforms.

Reply:
Ok, I see what you mean.
Please see my comment on your PR #3017

Reply:
Nit pick: Might be better if introduce PLATFORM_VENDOR or something similar, no need such large list of platform.

Reply:
Commit deleted.